_ruby_libs/text_rendering.rb

```diff
@@ -12,17 +13,34 @@ def rst_to_md(rst)
   end
 end
 
-# Modifies markdown image links so that they link to github user content
-def fix_image_links(text, raw_uri, additional_path = '')
+# Modifies markdown local links so that they link to github user content
+def fix_local_links(text, raw_uri, browse_uri)
   readme_doc = Nokogiri::HTML(text)
-  readme_doc.xpath("//img[@src]").each() do |el|
-    #puts 'img: '+el['src'].to_s
-    unless el['src'].start_with?('http')
-      el['src'] = ('%s/%s/' % [raw_uri, additional_path])+el['src']
+  readme_doc.css("img[src]").each() do |img|
+    begin
+      unless Addressable::URI.parse(img['src']).absolute?
+        img['src'] = raw_uri + "/" + img['src']
+      end
+    rescue Exception
```

Comment:
rescuing `Exception` is generally not wise in Ruby as interpreter/VM level failures can be represented as Exceptions. Leaving the `rescue` bare will default to rescuing from `StandardError` and its descendants which is probably what you want.

Ideally we would rescue only from the errors we expect to be raised but a bare rescue would be fine for now.

Reply:
Thanks for the detailed explanation, I replaced the `Exception` rescues with `InvalidURIError`, which is the exception that can be raised by `Addressable::URI.parse` on an invalid URI (as happened with our blown-up build in the past).